assets/shared/helpers/player/videopress-adapter.js

```diff
@@ -18,25 +18,42 @@ export const EMBED_PATTERN = /(videopress|video\.wordpress)\.com\/.+/i;
  */
 export const initializePlayer = ( element, w = window ) =>
 	new Promise( ( resolve ) => {
-		const onDurationChange = ( event ) => {
-			if (
-				event.source !== element.contentWindow ||
-				event.data.event !== 'videopress_durationchange' ||
-				! event.data.durationMs
-			) {
+		if ( element.dataset.duration ) {
```

Comment:
It's a fix to also resolve it when the duration was already set for other instances that can happen later.